internal/collector/gpu/gpu.go

```diff
@@ -0,0 +1,398 @@
+
+	return nil
+}
+
+func (c *Collector) Build(_ *slog.Logger, _ *mi.Session) error {
+	var err error
+
+	c.gpuEngineRunningTime = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "engine_time_seconds"),
+		"Total running time of the GPU in seconds.",
+		[]string{"pid", "phys", "eng", "engtype"},
+		nil,
+	)
+	c.gpuEngineUtilizationPercentage = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "utilization_percent"),
+		"GPU utilization percentage.",
+		[]string{"pid", "phys", "eng", "engtype"},
+		nil,
+	)
+
+	c.gpuAdapterMemoryDedicatedUsage = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "adapter_memory_dedicated_bytes"),
+		"Dedicated GPU memory usage in bytes.",
+		[]string{"phys"},
+		nil,
+	)
+	c.gpuAdapterMemorySharedUsage = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "adapter_memory_shared_bytes"),
+		"Shared GPU memory usage in bytes.",
+		[]string{"phys"},
+		nil,
+	)
+	c.gpuAdapterMemoryTotalCommitted = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "adapter_memory_committed_bytes"),
+		"Total committed GPU memory in bytes.",
+		[]string{"phys"},
+		nil,
+	)
+
+	c.gpuLocalAdapterMemoryUsage = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "local_adapter_memory_bytes"),
+		"Local adapter memory usage in bytes.",
+		[]string{"phys"},
+		nil,
+	)
+
+	c.gpuNonLocalAdapterMemoryUsage = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "non_local_adapter_memory_bytes"),
+		"Non-local adapter memory usage in bytes.",
+		[]string{"phys"},
+		nil,
+	)
+
+	c.gpuProcessMemoryDedicatedUsage = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "process_memory_dedicated_bytes"),
+		"Dedicated process memory usage in bytes.",
+		[]string{"pid", "phys"},
+		nil,
+	)
+	c.gpuProcessMemoryLocalUsage = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "process_memory_local_bytes"),
+		"Local process memory usage in bytes.",
+		[]string{"pid", "phys"},
+		nil,
+	)
+	c.gpuProcessMemoryNonLocalUsage = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "process_memory_non_local_bytes"),
+		"Non-local process memory usage in bytes.",
+		[]string{"pid", "phys"},
+		nil,
+	)
+	c.gpuProcessMemorySharedUsage = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "process_memory_shared_bytes"),
+		"Shared process memory usage in bytes.",
+		[]string{"pid", "phys"},
+		nil,
+	)
+	c.gpuProcessMemoryTotalCommitted = prometheus.NewDesc(
+		prometheus.BuildFQName(types.Namespace, Name, "process_memory_committed_bytes"),
+		"Total committed process memory in bytes.",
+		[]string{"pid", "phys"},
+		nil,
+	)
+
+	c.gpuEnginePerfDataCollector, err = pdh.NewCollector[gpuEnginePerfDataCounterValues](pdh.CounterTypeRaw, "GPU Engine", pdh.InstancesAll)
+	if err != nil {
+		return fmt.Errorf("failed to create GPU Engine perf data collector: %w", err)
```

Comment:
Please do not fail here.

See https://github.com/prometheus-community/windows_exporter/blob/5e1a8022377f5f8927e360ea6f622f43bbe99421/internal/collector/remote_fx/remote_fx.go#L235-L245

At least alle Performance Counter should be tried and one aggregated error should be returned.